### ISSUE TITLE
Tailscale re-authenticating daily

### ISSUE THREAD

**Issue by sellout** [tailscale/tailscale#5708]
### What is the issue?

I have only Tailscale SSH running on these boxes.
* `a`, Linux (NixOS), Tailscale 1.24.2
* `e`, Linux (Ubuntu), Tailscale 1.30.0
* `t`, MacOS, Tailscale 1.30.2

When I ssh from `t` to `e`, I get the message
```
# Tailscale SSH requires an additional check.
# To authenticate, visit: https://login.tailscale.com/a/cae2d9c2019d
```
after I authenticate, it seems to be valid for ~1 day before I need to re-authenticate. And my SSH sessions are usually started behind the scenes, like via TRAMP in Emacs, so it hangs for a while before I remember I should manually ssh to see if I need to re-auth.

Using ssh from `t` to `a`, I don’t have this problem. I’m trying to see if I run into it when sshing from `a` to `e`, but as that’s not a route I usually take, I’ll need to wait and see if it asks again tomorrow.

### Steps to reproduce

* ssh from `t` to `e`
* authenticate
* try again the next day

In practice, there are many SSH sessions each day, my MacOS laptop connects to various networks, disconnects, etc. So not sure if it’s really a “daily” thing, or if there’s one of these other factors causing me to need to auth again.

### OS

Linux, macOS

### OS version

NixOS 22.05, MacOS Monterey 12.6

### Tailscale version

1.30.0, 1.30.2

### Bug report

BUG-e671486f71ef5ae52de9f6a210001dfbcd11de8b9df90718e9795cf8f5cf2b05-20220921154154Z-fa3c120d70e29736

**Comment by DentonGentry** [comment]
You appear to have enabled Check mode for host `e`
https://tailscale.com/kb/1193/tailscale-ssh/#configure-tailscale-ssh-with-check-mode

**Comment by sellout** [comment]
Thanks, @DentonGentry!

According to that page, it looks like "check" is the default ACL for SSH, so that makes sense. I’ll edit that.

However, according to my services tab, `a` _is_ running Tailscale SSH, and is not running sshd (it also shows “SSH” on the machines tab). So now I’m confused as to why that one is _not_ asking me to check. Maybe check mode isn’t supported in 1.24.2?

The one sshd that’s remaining is on a computer I don’t have regular access to (`d`) and isn’t consistently online, so I don’t want to shut it down until I’m certain I’m in the vicinity to resolve any issues directly. And `d` (also on 1.24.2) does claim to be running Tailscale SSH as well.

The initials I’m using are the first letter of each actual host name, so hopefully you can see what’s what.

Thanks again for your help.

**Comment by DentonGentry** [comment]
The Services tab will show that there is an SSH service running on port 22. That can be OpenSSHd, anything listening on port 22 will be shown.

Tailscale SSH support was added in 1.26.0; tailscaled itself handles SSH to be able to authorize based on the SSH policy defined. You can see in the Machines list whether it has Tailscale SSH enabled:

<img width="264" alt="Screen Shot 2022-09-21 at 9 06 06 AM" src="https://user-images.githubusercontent.com/1140756/191555316-a690b004-6b2f-4315-8f56-f07c15d699cf.png">

("beachhouse" is aspirational by the way. It is neither a house nor is it on a beach)

**Comment by sellout** [comment]
Right, so my 1.24.2 machines do show SSH on the Machines list (and I’m certain I set them up with `--ssh`):
<img width="489" alt="Screen Shot 2022-09-21 at 10 42 22" src="https://user-images.githubusercontent.com/33031/191563272-dc1fee8a-86f9-4cb7-81af-ad13ebd3ba22.png">
And while the Services list does show any SSH service running on port 22, it seems to recognize and distinguish Tailscale SSH:
<img width="398" alt="Screen Shot 2022-09-21 at 10 43 03" src="https://user-images.githubusercontent.com/33031/191563536-375c81ad-b290-4407-a2e6-36bbfc018cac.png">
Finally, this looks like the commit that looks like it moved `tailscale up --ssh` out of WIP, and it’s first included in 1.24.0: https://github.com/tailscale/tailscale/commit/f0e2272e04c39210c7b35bb7e6bdb043346acd88

So, it still seems to me that I do have Tailscale SSH running on the hosts with 1.24.2, but they’re (apparently) not respecting the `"action": "check",` part of the ACL.

I’m going to try to get those boxes updated (they’re both on NixOS, so I think I just need to grab the tailscale from unstable instead of the 22.05 release.

**Comment by sellout** [comment]
Sorry, forgot to follow-up. I upgrade `a` to 1.30.0, and it started asking me for daily auth just like `e` was. So then I went and changed the ACL from `"check"` to `"accept"` (using GitOps, yay!) and that made it stop asking (as expected). So, there is no bug, and my initial confusion stemmed from the differences between 1.24 and 1.30 behavior.

Thanks for the help!
